### ISSUE TITLE
error on DELETE of image versions on registry v2

### ISSUE THREAD

**Issue by vmalaga** [tsuru/tsuru#1361]
Hi.

we are using docker registry v2 for months and searching for the reason of fast growing on the registry strorage and looking at registry logs, I see the DELETE requests on registry are being made with **v1** at the begining of the url and the registry always return 404

```url
"DELETE /v1/repositories/tsuru/app-lander-beta:v154/ HTTP/1.1" 404 19 "" "Go-http-client/1.1"
```
however if the deletion is done with **v2** the registry find the image and can delete it.

Are you going to change this or can I set it somewhere?

**Comment by tarsisazevedo** [comment]
I think the problem is [this line](https://github.com/tsuru/docker-cluster/blob/a03397cb061d7df50c8601904b7ca55efc156b0d/cluster/image.go#L94) at docker-cluster.

To fix that we can try to delete first on v2 registry, and if it fails, try to v1.

**Comment by vmalaga** [comment]
I think with the new v2 api can't DELETE by tags, need the digest or manifests

https://docs.docker.com/registry/spec/api/
https://github.com/docker/distribution/blob/master/docs/spec/api.md#deleting-an-image

**Comment by vmalaga** [comment]
Hi @tarsisazevedo when is going out this feature, I see you say with 1.0 milestone, but it take's a long time.

**Comment by tarsisazevedo** [comment]
Hi @vmalaga, we decide to normalize tsuru api before release 1.0 and it takes a while to be done, sorry.
We have a release candidate package on `ppa:tsuru/rc` with this issue included, and we are about to release oficial package next week.

**Comment by vmalaga** [comment]
Hi @tarsisazevedo 

This night i have updated our tsuru instance but the DELETE on registry is still requested on v1 api and return a 404 error

```
registry_v2          | 172.16.70.100 - - [04/Oct/2016:15:07:10 +0000] "DELETE /v1/repositories/tsuru/app-varnishapi:v10/ HTTP/1.1" 404 19 "" "Go-http-client/1.1"
```
is this issue fixed on tsuru 1.0 ?

**Comment by andrewsmedina** [comment]
this issues was fixed in tsuru 1.0

**Comment by vmalaga** [comment]
Hi @andrewsmedina we are now with the last version of tsuru and the log is from today

```
root@tsurudev:~# tsurud version
tsurud version 1.1.0.
```

although in the log I see v2 calls has also calls to v1 with 404 reponses  trying to delete tags layers. send you the log
[registry.txt](https://github.com/tsuru/tsuru/files/510968/registry.txt)
